src/lib.js

```diff
@@ -68,42 +94,37 @@ async function getOmgevingInfo(data) {
   return data;
 }
 
-
-const callchain = [
-  wrapApiCall(getFullObjectData),
-  wrapApiCall(getOmgevingInfo)
-]
-
-//user-provided function for featureQuery to format request URL
-//featureQuery will call this function with the following arguments:
-// baseUrl string: the base url of the API (from config)
-// xy: the x,y of the clicked point on the map (longitude, latitude).
 function requestFormatter(baseUrl, xy) {
   let xyRD = transformCoords.forward(xy);
   return `${baseUrl}${xyRD.x},${xyRD.y},50`
 }
 
-//user-provided function for featureQuery to parse Ajax response
-//we need:
-//* query (lat/lon)
-//* object
-//* omgeving:
-//  * dichtsbijzijnd adres
-//  * buurtnaam
-//  * buurtcode
-//  * stadsdeelnaam
-//  * stadsdeelcode
-//  * buurtcombinatienaam
-//  * buurtcombinatiecode
-//
-
-//So we need several of these request/responseformatter things.
-//And then we chain them together.
 function responseFormatter(res) {
-  let filtered = res.results.filter(x => x.hoofdadres === true);
+  let filtered;
+  try {
+    filtered = res.results.filter(x => x.hoofdadres === true);
+  } catch (e) {
+    throw {
+      error: 'no results property found on query response.',
+      response: res,
+      originalError: e
+    }
+  }
   return filtered.length > 0 ? filtered[0] : null;
 }
 
-export { callchain, getFullObjectData, getOmgevingInfo, requestFormatter, responseFormatter };
 
 
+
+async function pointQueryChain (click) {
+  try {
+    const result = await getBagInfo(click) 
+    .then(getFullObjectData)
+    .then(getOmgevingInfo);
+    return result;
```

Comment:
Kan je hier niet zeggen:
```
const bagInfo = await getBagInfo(click);
const fullObjectData = await getFullObjectData(bagInfo);
return await getOmgevingInfo(fullObjectData);
```

Reply:
dat zou misschien kunnen, maar daarmee wordt het mijns inziens niet duidelijker;
` getBagInfo then getFullObjectData then getOmgevingsInfo ` is op zich precies wat er moet gebeuren en zo staat het dan ook in code